x-pack/solutions/security/plugins/security_solution/public/detections/components/attacks/table/attack_details/attack_details_container.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { AttackDiscoveryAlert } from '@kbn/elastic-assistant-common';
+import type { Filter } from '@kbn/es-query';
+import { EuiSpacer, EuiTabs, EuiTab } from '@elastic/eui';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { AlertsTab } from './alerts_tab';
+import * as i18n from './translations';
+
+/** Tab ID constant for the Attack Summary tab */
+export const ATTACK_SUMMARY_TAB = 'attackSummaryTab';
+/** Tab ID constant for the Alerts tab */
+export const ALERTS_TAB = 'alertsTab';
+
+/** Test subject constant for the tabs container */
+export const TABS_TEST_ID = 'tabs';
+
+interface TabInfo {
+  content: JSX.Element;
+  id: string;
+  name: string;
+}
+
+interface Props {
+  /** The attack discovery alert document. If undefined, only the Alerts tab will be shown. */
+  attack?: AttackDiscoveryAlert;
+  /** Whether to show anonymized values instead of replacements */
+  showAnonymized?: boolean;
+  /** Filters applied from grouping */
+  groupingFilters: Filter[];
+  /** Default filters to apply to the alerts table */
+  defaultFilters: Filter[];
+  /** Whether the alerts table is in a loading state */
+  isLoading: boolean;
+}
+
+/**
+ * Main component that provides a container interface for viewing attack summary
+ * and associated alerts when an attack group is selected in the attacks table.
+ * Manages tab selection state and resets to first tab when the attack changes.
+ * If attack is undefined, only the Alerts tab will be rendered.
+ */
+export const AttackDetailsContainer = React.memo<Props>(
+  ({ attack, groupingFilters, defaultFilters, isLoading }) => {
+    const tabs = useMemo<TabInfo[]>(() => {
+      const tabsList: TabInfo[] = [];
+
+      if (attack) {
+        tabsList.push({
+          id: ATTACK_SUMMARY_TAB,
+          name: i18n.ATTACK_SUMMARY,
+          content: (
+            <>
+              <EuiSpacer />
+              {/* TODO: Placeholder for `AttackSummaryTab` */}
+              <div data-test-subj="attackSummaryTabPlaceholder" />
+            </>
+          ),
+        });
+      }
+
+      tabsList.push({
+        id: ALERTS_TAB,
+        name: i18n.ALERTS,
+        content: (
+          <>
+            <EuiSpacer />
+            <AlertsTab
+              groupingFilters={groupingFilters}
+              defaultFilters={defaultFilters}
```

Comment:
consider here also combining `groupingFilters` and `defaultFilters` into a single `filters` prop?